src/middleware/cloudWatch.ts

```diff
@@ -0,0 +1,59 @@
+// To add logs to cloudwatch
+// Custom middlware: https://github.com/middyjs/middy#configurable-middlewares
+
+import { getData, WrappedDocument } from "@govtechsg/open-attestation";
+import middy, { MiddlewareObj } from "@middy/core";
+import { NotarisationResult } from "src/functionHandlers/notarisePdt/handler";
+import { HealthCertDocument, Observation } from "src/types";
+import { logError, trace } from "./trace";
+
+// export const { trace } = getLogger("cloudWatchMiddleware");
+export type Request = Pick<middy.Request, "event" | "response">;
+
+// Custom middleware must return an obhect of { before?: fn, after?: fn, onError?: fn }
+// Note: Do not return a response as
+// this will totally stop the execution of successive middlewares in any phase (before, after, onError) and returns an early response
+// https://github.com/middyjs/middy#interrupt-middleware-execution-early
+
+export class CloudWatchMiddleware
+  implements Pick<MiddlewareObj, "before" | "after">
+{
+  provider = "";
+
+  before = async (req: Request): Promise<void> => {
+    const wrappedDocument = req.event
+      .body as WrappedDocument<HealthCertDocument>;
+    const data = getData(wrappedDocument);
+    const provider: string = data.issuers[0].name;
```

Comment:
hmmm

```
 "issuers": [
      {
        "name": "51eb71d5-ee7a-4503-bf9c-d399f89874d8:string:MINISTRY OF HEALTH (SINGAPORE)",
        "id": "2ec02df6-72c2-40d5-a03d-d2a6876dcf57:string:did:ethr:0x82948a537e886Ea8ffF5Bd29DDA95224bF74c35F",
        "revocation": {
          "type": "6e7d3944-e27c-430e-b13a-b1b343adf99c:string:NONE"
        },
        "identityProof": {
          "type": "2555f9ca-5f3a-4508-b248-e8e80cf5027b:string:DNS-DID",
          "location": "0ea88c24-3547-4b9d-a4b0-8516732c09dd:string:moh.gov.sg",
          "key": "f71e2ffd-86a1-488d-829c-0298c228ce85:string:did:ethr:0x82948a537e886Ea8ffF5Bd29DDA95224bF74c35F#controller"
        }
      }
    ],
```

not sure which value we should take here, am conflicted between issuers[0].name and issuers.identityProof.location

for whitelisting we check location right?